File: languages/django_models/creature.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Creature(AbstractElementModel):

    # Biology
    appearance = models.TextField(blank=True, null=True)
    weight = models.PositiveIntegerField(blank=True, null=True)
    height = models.PositiveIntegerField(blank=True, null=True)
    species = models.ManyToManyField("Species", blank=True, related_name="creature_species")

    # Behaviour
    habits = models.TextField(blank=True, null=True)
    demeanor = models.TextField(blank=True, null=True)
    traits = models.ManyToManyField("Trait", blank=True, related_name="creature_traits")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="creature_abilities")
    languages = models.ManyToManyField("Language", blank=True, related_name="creature_languages")

    # World
    status = models.TextField(blank=True, null=True)
    birth_date = models.PositiveIntegerField(blank=True, null=True)
    location = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="creature_location")
    zone = models.ForeignKey("Zone", on_delete=models.SET_NULL, blank=True, null=True, related_name="creature_zone")

    # Ttrpg
    challenge_rating = models.PositiveIntegerField(blank=True, null=True)
    hit_points = models.PositiveIntegerField(blank=True, null=True)
    armor_class = models.PositiveIntegerField(blank=True, null=True)
    speed = models.PositiveIntegerField(blank=True, null=True)
    actions = models.ManyToManyField("Ability", blank=True, related_name="creature_actions")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Creature.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Creature : Element
{
    [JsonProperty("appearance"), TextAttribute("")]
    public string appearance;
    [JsonProperty("weight"), Integer(0)]
    public int weight;
    [JsonProperty("height"), Integer(0)]
    public int height;
    [JsonProperty("species"), ReferenceAttribute(typeof(Species), true)]
    public string species;
    [JsonProperty("habits"), TextAttribute("")]
    public string habits;
    [JsonProperty("demeanor"), TextAttribute("")]
    public string demeanor;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
    [JsonProperty("status"), TextAttribute("")]
    public string status;
    [JsonProperty("birth_date"), Integer(0)]
    public int birthDate;
    [JsonProperty("location"), ReferenceAttribute(typeof(Location))]
    public string location;
    [JsonProperty("zone"), ReferenceAttribute(typeof(Zone))]
    public string zone;
    [JsonProperty("challenge_rating"), Integer(0)]
    public int challengeRating;
    [JsonProperty("hit_points"), Integer(0)]
    public int hitPoints;
    [JsonProperty("armor_class"), Integer(0)]
    public int armorClass;
    [JsonProperty("speed"), Integer(0)]
    public int speed;
    [JsonProperty("actions"), ReferenceAttribute(typeof(Ability), true)]
    public string actions;
}
